src/modules/feedback-reports/index.ts

```diff
@@ -0,0 +1,264 @@
+            )
+    );
+
+    if (modifiedCount === 0)
+        throw createHttpError(
+            401,
+            'You must be the creator in order to modify the feedback report'
+        );
+}
+
+// TODO: extend this to write to a trash collection rather than actually delete
+/**
+ * @description Deletes a feedback report by Id
+ * @param {string} reportId - Id of the feedback report to delete
+ * @returns {Promise} Void promise
+ * @throws {Error} If provided report id is invalid or if specified feedback report to delete does not exist
+ */
+export async function deleteFeedbackReport(reportId: string) {
+    if (!ObjectID.isValid(reportId))
+        throw createHttpError(400, 'Invalid report id provided');
+
+    const { deletedCount } = await useCollection(
+        'FeedbackReports',
+        (FeedbackReports) =>
+            FeedbackReports.deleteOne({ _id: new ObjectID(reportId) })
+    );
+
+    if (deletedCount === 0)
+        throw createHttpError(404, 'Feedback report not found');
+}
+
+/**
+ * @description Sets the resolved attribute of a feedback report to the new resolved status provided
+ * @param {string} reportId - Id of the report
+ * @param {boolean} newResolvedStatus - New resolved status
+ * @param {User} user - User object of updater
+ * @returns {Promise} Void promise
+ * @throws {Error} If provided report id is invalid or if unable to update the status of the specified feedback report
+ */
+export async function updateResolvedStatus(
+    reportId: string,
+    newResolvedStatus: boolean,
+    user: User
+) {
+    if (!ObjectID.isValid(reportId))
+        throw createHttpError(400, 'Invalid report id provided');
+
+    const { modifiedCount } = await useCollection(
+        'FeedbackReports',
+        (FeedbackReports) =>
+            FeedbackReports.updateOne(
+                {
+                    _id: new ObjectID(reportId),
+                },
+                {
+                    $set: {
+                        'meta.updatedAt': new Date(),
+                        'meta.updatedBy.by': user,
+                        resolved: newResolvedStatus,
+                    },
+                },
+                {
+                    upsert: false,
+                }
+            )
+    );
+
+    if (modifiedCount === 0)
+        throw createHttpError(
+            400,
+            'Could not update resolved status of feedback report'
+        );
+}
+
+/**
+ * @description Adds a reply to feedback a report
+ * @param {User} user - User object of the replier
+ * @param {string} reportId - Id of the report
+ * @param {string} content - Content of the reply
+ * @returns {Promise} Void promise
+ * @throws {Error} If provided report id is invalid or if unable to add reply to feedback report
+ */
+export async function replyToFeedbackReport(
+    user: User,
+    reportId: string,
+    content: string
+) {
+    if (!ObjectID.isValid(reportId))
+        throw createHttpError(400, 'Invalid report id provided');
```

Comment:
I think these should probably be 404's.  It's not found because the id provided is invalid, but at the same time there's nothing a user can do to fix it.  400's are meant when there's an input error the user can fix.

Also, it looks like you do the validation for ObjectID's at the middleware level, so these may be unnecessary.